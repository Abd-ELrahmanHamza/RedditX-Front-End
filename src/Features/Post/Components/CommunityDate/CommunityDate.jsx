// Import bootstrap components
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import Tooltip from "react-bootstrap/Tooltip";

// Import icons
import { FiEyeOff } from "react-icons/fi";
import { TbCake } from "react-icons/tb";

// import SaveChangesModal from "c:/Users/Lenovo/Downloads/Front-End-main/Front-End-main/src/Features/Subreddit/Components/SaveChangesModal/SaveChangesModal";
import {
  BirthDay,
  Private,
  Description,
  Title,
  CommunityImage,
  CommunityName,
} from "./CommunityDate.styled";

// Import contexts
import { useSubmitDestination } from "Features/Post/Contexts/submitDestination";

// Import utils
import getDaysCount from "Features/Post/Utils/getDaysCount";
import { BASE_URL } from "API/axios";
/**
 *
 * @returns {React.Component} CreatePost component
 */
const CommunityDate = ({ communityInfo }) => {
  return (
    <>
      <Title>
        <CommunityImage
          crossOrigin="anonymous"
          src={`${BASE_URL}/subreddits/files/${communityInfo.icon}`}
          alt="community"
          as={`${communityInfo.banner ? "img" : "div"}`}
        />
        {communityInfo && (
          <CommunityName data-testid="community-name">
            r/{communityInfo._id.substring(3)}
          </CommunityName>
        )}
      </Title>
      {communityInfo && (
        <Description data-testid="community-description">
          {communityInfo.description}
        </Description>
      )}
      <OverlayTrigger
        placement={"bottom"}
        overlay={
          <Tooltip>
            {getDaysCount(new Date(), communityInfo.createdAt) +
              ` day ${
                getDaysCount(new Date(), communityInfo.createdAt) > 1 ? "s" : ""
              } ago`}
          </Tooltip>
        }
        delay="200"
      >
        <BirthDay>
          <div className="cake-day">
            <span className="icon">
              <TbCake />
            </span>

            <span className="text">
              Created{" "}
              {new Date(communityInfo.createdAt).toDateString().substring(4)}
            </span>
          </div>
        </BirthDay>
      </OverlayTrigger>

      <Private>
        <div className="container">
          <OverlayTrigger
            placement={"top"}
            overlay={
              <Tooltip>
                Only approved users can view and submit to this community
              </Tooltip>
            }
            delay="200"
          >
            <span className="icon">
              <FiEyeOff />
            </span>
          </OverlayTrigger>
          <span className="text">Private</span>
        </div>
      </Private>
    </>
  );
};

export default CommunityDate;
