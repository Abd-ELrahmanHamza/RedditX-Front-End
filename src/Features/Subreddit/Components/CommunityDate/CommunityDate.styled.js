import styled from "styled-components";

export const AddDescription = styled.div`
  transition: all 0.1s linear 0s;
  display: block;
  background-color: #f6f7f8;
  border-radius: 4px;
  padding: 8px;
  margin-bottom: 12px;
  margin-top: 8px;
  border: 1px solid #dae0e6;
  cursor: pointer;

  div {
    font-size: 12px;
    font-weight: 700;
    line-height: 16px;
    color: #0079d3;
  }

  :hover {
    border: 1px solid #0079d3;
  }
`;

export const BirthDay = styled.div`
  margin-top: 12px;

  .cake-day {
    font-family: "Noto Sans", Arial, sans-serif;
    font-size: 14px;
    font-weight: 400;
    line-height: 18px;
    display: flex;
    flex-flow: row nowrap;
  }

  .icon {
    font-size: 20px;
    font-weight: 400;
    height: 20px;
    line-height: 20px;
    vertical-align: middle;
    width: 20px;
    margin: -2px 8px 0 0;
  }

  .text {
    color: #7c7c7c;
  }
`;

export const Private = styled.div`
  margin-top: 8px;

  div.container {
    padding: 0;
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
  }
  span.icon {
    height: 16px;
    padding-right: 4px;
    vertical-align: top;
    font-size: 20px;
    font-weight: 400;
    line-height: 20px;
    width: 20px;
  }

  span.text {
    font-family: "Noto Sans", Arial, sans-serif;
    font-size: 14px;
    font-weight: 400;
    line-height: 18px;
  }
`;
