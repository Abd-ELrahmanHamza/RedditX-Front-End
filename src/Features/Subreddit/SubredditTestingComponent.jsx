import { ThemeProvider } from "styled-components";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";
import { BrowserRouter as Router } from "react-router-dom";
import { SubRedditProvider } from "Features/Subreddit/Contexts/SubRedditProvider";
import { SubRedditIDProvider } from "./Contexts/SubRedditIDProvider";
import { IsModeratorProvider } from "Features/Subreddit/Contexts/IsModeratorProvider";
import { IsBannedProvider } from "Features/Subreddit/Contexts/IsBannedProvider";
import { IsMutedProvider } from "Features/Subreddit/Contexts/IsMutedProvider";

const theme = { ...defaultTheme, ...lightTheme };

/**
 * Function that wraps Children props inside a Browser Router and a Theme Provider
 * @returns {React.Component}The component is being returned.
 */
const SubredditTestingComponent = ({ children }) => {
  return (
    <Router>
      <ThemeProvider theme={theme}>
        <SubRedditProvider>
          <SubRedditIDProvider>
            <IsModeratorProvider>
              <IsBannedProvider>
                <IsMutedProvider>{children}</IsMutedProvider>
              </IsBannedProvider>
            </IsModeratorProvider>
          </SubRedditIDProvider>
        </SubRedditProvider>
      </ThemeProvider>
    </Router>
  );
};

export default SubredditTestingComponent;
