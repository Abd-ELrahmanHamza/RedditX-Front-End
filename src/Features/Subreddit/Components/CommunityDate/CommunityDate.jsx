import { FiEyeOff } from "react-icons/fi";
import { TbCake } from "react-icons/tb";
import { AddDescription, BirthDay, Private } from "./CommunityDate.styled";

const CommunityDate = () => {
  let isPrivate = true;
  return (
    <>
      <AddDescription>
        <div>Add description</div>
      </AddDescription>
      <BirthDay>
        <div className="cake-day">
          <span className="icon">
            <TbCake />
          </span>
          <span className="text">Created date</span>
        </div>
      </BirthDay>
      {isPrivate && (
        <Private>
          <div className="container">
            <span className="icon">
              <FiEyeOff />
            </span>
            <span className="text">Private</span>
          </div>
        </Private>
      )}
    </>
  );
};

export default CommunityDate;
