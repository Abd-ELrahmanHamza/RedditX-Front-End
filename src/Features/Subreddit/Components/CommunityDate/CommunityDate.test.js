import { render, screen } from "@testing-library/react";
import CommunityDate from "./CommunityDate";
import SubredditTestingComponent from "Features/Subreddit/SubredditTestingComponent";

// Enzyme.configure({ adapter: new Adapter() });

describe("Description and created date of subreddit", () => {
  it("description should appear", () => {
    //Arrange
    render(
      <SubredditTestingComponent>
        <CommunityDate />
      </SubredditTestingComponent>
    );
  });
});
