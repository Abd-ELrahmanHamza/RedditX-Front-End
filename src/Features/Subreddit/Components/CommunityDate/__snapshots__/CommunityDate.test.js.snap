// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description and created date of subreddit should render without crashing 1`] = `
<Fragment>
  <Description
    onClick={[Function]}
  />
  <OverlayTrigger
    defaultShow={false}
    delay="200"
    overlay={
      <Tooltip
        placement="right"
      >
        16 days ago
      </Tooltip>
    }
    placement="bottom"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <styled.div>
      <div
        className="cake-day"
      >
        <span
          className="icon"
        >
          <TbCake />
        </span>
        <span
          className="text"
        >
          Created Oct 17, 2022
        </span>
      </div>
    </styled.div>
  </OverlayTrigger>
  <styled.div>
    <div
      className="container"
    >
      <OverlayTrigger
        defaultShow={false}
        delay="200"
        overlay={
          <Tooltip
            placement="right"
          >
            Only approved users can view and submit to this community
          </Tooltip>
        }
        placement="top"
        trigger={
          Array [
            "hover",
            "focus",
          ]
        }
      >
        <span
          className="icon"
        >
          <FiEyeOff />
        </span>
      </OverlayTrigger>
      <span
        className="text"
      >
        Private
      </span>
    </div>
  </styled.div>
  <SaveChangesModal
    onDiscard={[Function]}
    onHide={[Function]}
    onSave={[Function]}
    show={false}
  />
</Fragment>
`;
